### PR TITLE
refactor(konnect): add enqueueObjectForKonnectGatewayControlPlane to generically list objects that refer to KonnectGatewayControlPlane

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

Generically list objects that refer to `KonnectGatewayControlPlane`.

### CODE REVIEW

File: controller/konnect/constraints/constraints.go [hunk 1]
```diff
@@ -75,3 +75,14 @@ type SupportedKonnectEntityPluginReferenceableType interface {
...
+
+// ClientObject represents an object that has a name and a namespace
+// (might be empty for cluster-scoped objects).
+type ClientObject[T any] interface {
+	*T
+
+	GetName() string
+	GetNamespace() string
+
+	client.Object
```
Comment (czeslavo) [inline controller/konnect/constraints/constraints.go]: Doesn't `client.Object` already provide `GetName` and `GetNamespace`?

Comment (czeslavo) [reply controller/konnect/constraints/constraints.go]: BTW Wouldn't it be possible to omit this interface and use the existing `EntityType`?

Comment (pmalek) [reply controller/konnect/constraints/constraints.go]: Ah yes 🤦 27e38911dad8aee4c5b3fbab88d93c5900742655

Comment (pmalek) [reply controller/konnect/constraints/constraints.go]: > BTW Wouldn't it be possible to omit this interface and use the existing EntityType?

This way it's less restrictive and can be used with other types, while `EntityType` already has some assumptions about konnect related methods that this type has to have.

Comment (czeslavo) [reply controller/konnect/constraints/constraints.go]: I agree it's less restrictive, but we only use it now for Konnect entities (namely, in `enqueueObjectForKonnectGatewayControlPlane`, which implies a Konnect entity use case). We might as well keep it simple, not introduce more types, and add it when we find it required.

Comment (pmalek) [reply controller/konnect/constraints/constraints.go]: 👍 f382e23f8fd1547593f6371f3b25777e01746034

**Review by czeslavo — APPROVED** [review]

**Pull request merged**